Src/FluentAssertions/Primitives/StringEqualityStrategy.cs

```diff
@@ -84,4 +115,122 @@ public string ExpectationDescription
+        string prefix, string text, int indexOfStartingPhrase, string suffix)
+    {
+        var subjectLength = GetLengthOfPhraseToShowOrDefaultLength(text[indexOfStartingPhrase..]);
+        const char ellipsis = '\u2026';
+
+        stringBuilder.Append(prefix);
+
+        if (indexOfStartingPhrase > 0)
+        {
+            stringBuilder.Append(ellipsis);
+        }
+
+        stringBuilder.Append(text
+            .Substring(indexOfStartingPhrase, subjectLength)
+            .Replace("\r", "\\r", StringComparison.OrdinalIgnoreCase)
+            .Replace("\n", "\\n", StringComparison.OrdinalIgnoreCase));
+
+        if (text.Length > (indexOfStartingPhrase + subjectLength))
+        {
+            stringBuilder.Append(ellipsis);
+        }
+
+        stringBuilder.AppendLine(suffix);
+    }
+
+    /// <summary>
+    /// Calculates the start index of the visible segment from <paramref name="value"/> when highlighting the difference at <paramref name="indexOfFirstMismatch"/>.
+    /// </summary>
+    /// <remarks>
+    /// Either keep the last 10 characters before <paramref name="indexOfFirstMismatch"/> or a word begin (separated by whitespace) between 15 and 5 characters before <paramref name="indexOfFirstMismatch"/>.
+    /// </remarks>
+    private static int GetStartIndexOfPhraseToShowBeforeTheMismatchingIndex(string value, int indexOfFirstMismatch)
+    {
+        const int defaultCharactersToKeep = 10;
+        const int minCharactersToKeep = 5;
+        const int maxCharactersToKeep = 15;
+        const int lengthOfWhitespace = 1;
+
+        if (indexOfFirstMismatch <= defaultCharactersToKeep)
+        {
+            return 0;
+        }
+
+        var indexToStartSearchingForWordBoundary = Math.Max(indexOfFirstMismatch - (maxCharactersToKeep + lengthOfWhitespace), 0);
+
+        var indexOfWordBoundary = value
+            .Substring(indexToStartSearchingForWordBoundary, (maxCharactersToKeep - minCharactersToKeep) + lengthOfWhitespace)
+            .IndexOf(' ', StringComparison.OrdinalIgnoreCase);
+
+        if (indexOfWordBoundary >= 0)
+        {
+            return indexToStartSearchingForWordBoundary + indexOfWordBoundary + lengthOfWhitespace;
+        }
+
+        return indexOfFirstMismatch - defaultCharactersToKeep;
+    }
+
+    /// <summary>
+    /// Calculates how many characters to keep in <paramref name="value"/>.
+    /// </summary>
+    /// <remarks>
+    /// If a word end is found between 15 and 25 characters, use this word end, otherwise keep 20 characters.
+    /// </remarks>
+    private static int GetLengthOfPhraseToShowOrDefaultLength(string value)
+    {
+        const int defaultLength = 20;
+        const int minLength = 15;
+        const int maxLength = 25;
+        const int lengthOfWhitespace = 1;
+
+        var indexOfWordBoundary = value[..Math.Min(maxLength + lengthOfWhitespace, value.Length)]
+            .LastIndexOf(' ');
```

Comment:
It's because in this overload the parameter is the 0-based [starting position of the search](https://learn.microsoft.com/en-us/dotnet/api/system.string.lastindexof?view=net-6.0#system-string-lastindexof(system-char-system-int32)).

Reply:
I did figure out why `-1` was necessary for the statement to be equivalent.
It's whether that was desirable I didn't get to look into.

Reply:
Fixed in [110ed76](https://github.com/fluentassertions/fluentassertions/pull/2307/commits/110ed76ff676dace0b57314fc71bd65f28abf5fe).